src/libraries/System.Runtime.Numerics/tests/BigInteger/BigIntegerToStringTests.cs

```diff
@@ -439,6 +439,38 @@ public static void CustomFormatPerMille()
             RunCustomFormatToStringTests(s_random, "#\u2030000000", CultureInfo.CurrentCulture.NumberFormat.NegativeSign, 6, PerMilleSymbolFormatter);
         }
 
+        public static IEnumerable<object[]> RunFormatScientificNotationToBigIntegerAndViceVersaData()
+        {
+            yield return new object[] { "1E+1000", "1E+1000" };
+            yield return new object[] { "1E+1001", "1E+1001" };
```

Comment:
@tannergooding I was not able to get a test with "1E+2147483639" to work locally, I think it ran for too long.

Reply:
I think that's fine, as long as we have some basic tests that values over 1E+1000 work it should be good enough

Testing the extreme edge cases is going to run into lots of issues.

Reply:
I added a couple bigger test cases.